src/JMS/Serializer/XmlDeserializationVisitor.php

```diff
@@ -151,6 +151,28 @@ public function visitDouble($data, array $type, Context $context)
 
     public function visitArray($data, array $type, Context $context)
     {
+        $metadata = $this->currentMetadata;
+        if (null === $metadata && $context->getMetadataStack()->count()) {
```

Comment:
instead of this "dirty" check, what about adding here a `setCurrentMetadata` call as [here](https://github.com/schmittjoh/serializer/blob/f1d9f1662c4a9b169fd0c81f674187d0032c442c/src/JMS/Serializer/XmlDeserializationVisitor.php#L265). Should be faster and "cleaner"